public/components/custom_panels/helpers/utils.tsx

```diff
@@ -9,6 +9,120 @@
  * GitHub history for details.
  */
 
-export const isNameValid = (name:string)=>{
-    return (name.length >= 50 || name.length === 0) ? false : true;
-}
\ No newline at end of file
+import dateMath from '@elastic/datemath';
+import _ from 'lodash';
+import {
+  DATE_FORMAT,
+  INDEX_REGEX,
+  VisualizationType,
+} from '../../../../common/constants/custom_panels';
+import PPLService from '../../../services/requests/ppl';
+
+/*
+ * "Utils" This file contains different reused functions in operational panels
+ */
+
+// Name validation 0>Name<=50
+export const isNameValid = (name: string) => {
+  return name.length >= 50 || name.length === 0 ? false : true;
+};
+
+// DateTime convertor to required format
+export const convertDateTime = (datetime: string, isStart = true) => {
+  if (isStart) return dateMath.parse(datetime).format(DATE_FORMAT);
+  return dateMath.parse(datetime, { roundUp: true }).format(DATE_FORMAT);
+};
+
+// Builds Final Query by adding time and query filters to the original visualization query
+// -> Final Query is as follows:
+// -> finalQuery = indexPartOfQuery + timeQueryFilter + panelFilterQuery + filterPartOfQuery
+// -> finalQuery = source=opensearch_dashboards_sample_data_flights
+//                  + | where utc_time > timestamp(‘2021-07-01 00:00:00’) and utc_time < timestamp(‘2021-07-02 00:00:00’)
+//                  + | where Carrier='OpenSearch-Air'
+//                  + | stats sum(FlightDelayMin) as delays by Carrier
+const queryAccumulator = (
+  originalQuery: string,
+  timestampField: string,
+  startTime: string,
+  endTime: string,
+  panelFilterQuery: string
+) => {
+  const indexMatchArray = originalQuery.match(INDEX_REGEX);
+  if (indexMatchArray == null) {
+    throw Error('index not found in Query');
+  }
+  const indexPartOfQuery = indexMatchArray[0];
+  const filterPartOfQuery = originalQuery.replace(INDEX_REGEX, '');
+  const timeQueryFilter = ` | where ${timestampField} > timestamp('${convertDateTime(
```

Comment:
probably won't make a difference in most cases, but should we make the time range inclusive (e.g. `>=` and `<=`)?

Reply:
I can make this inclusive on both ends `>= and <=`. Let's make this consistent across modules in observability. Looping in Eric. @mengweieric what's your opinion on this?

Reply:
Yes, inclusive makes sense here.